src/hotspot/cpu/x86/x86.ad

```diff
@@ -6959,8 +6948,11 @@ instruct vcastStoX(vec dst, vec src, vec vtmp, rRegP scratch) %{
 %}
 
 instruct vcastStoX_evex(vec dst, vec src) %{
-  predicate((UseAVX > 2 && VM_Version::supports_avx512vlbw()) ||
-            (Matcher::vector_length_in_bytes(n) >= Matcher::vector_length_in_bytes(n->in(1)))); // dst >= src
+  predicate((Matcher::vector_element_basic_type(n) == T_BYTE && UseAVX > 2 && VM_Version::supports_avx512vlbw()) ||
```

Comment:
The following code is redundant:
```
  switch (to_elem_bt) {
      case T_BYTE:
        if (!VM_Version::supports_avx512vl()) {
          vlen_enc = Assembler::AVX_512bit;
        }
```

Reply:
Yes it is. However, the intention of this code seems that `vpmovwb` is promoted to `AVX_512bit` when avx512vl is not available, and the predicate is too restrictive. I think this needs a closer look.

Reply:
AVX512VL support always comes along with AVX512BW in Intel CPUs. Xeon Phis didn't support VL, but they didn't support BW as well. So, it's safe to assume that BW implies VL and use `VM_Version::supports_avx512vlbw()` instead of `VM_Version::supports_avx512bw()`. It makes the aforementioned code effectively dead.